lib/tasks/scripts/telemedicine_reports.rb

```diff
@@ -0,0 +1,322 @@
+            ]
+
+            district[:facilities].each do |facility|
+              csv << [
+                "",
+                "",
+                facility[:name],
+                "",
+                "",
+                facility[:users],
+                "",
+                facility[:telemed_data][:visits],
+                facility[:telemed_data][:high_bp],
+                facility[:telemed_data][:high_bs],
+                facility[:telemed_data][:high_bp_or_bs],
+                "",
+                ""
+              ]
+            end
+          end
+        end
+
+        csv << []
+        csv << []
+
+        daily_activity_data = raw_mixpanel_data.group_by { |row| row[:date] }.sort_by { |date, _rows| date }.map { |date, rows|
+          [date.strftime("%d %b %Y"), rows.uniq { |row| row[:user_id] }.count, sum_values(rows, :clicks)]
+        }
+
+        csv << ["Date", "Unique users", "Total TC requests"]
+        daily_activity_data.each do |row|
+          csv << row
+        end
+      end
+    end
+
+    private
+
+    def format_mixpanel_data(mixpanel_data)
+      mixpanel_data.group_by { |row| row[:state] }.map { |state, districts|
+        {state: state,
+         clicks: sum_values(districts, :clicks),
+         districts: districts.group_by { |row| row[:district] }.map { |district, users|
+                      {district: district,
+                       clicks: sum_values(users, :clicks)}
+                    }}
+      }
+    end
+
+    def format_facility_data(facility_data)
+      facility_data.group_by { |facility| facility[:state] }.map { |state, state_facilities|
+        {state: state,
+         count: state_facilities.count,
+         hwc_and_sc_count: hwc_and_sc_count(state_facilities),
+         telemed_data: calculate_aggregates(state_facilities),
+         users: sum_values(state_facilities, :users),
+         districts: state_facilities.group_by { |facility| facility[:district] }.map { |district, district_facilities|
+                      {district: district,
+                       state: state,
+                       count: district_facilities.count,
+                       hwc_and_sc_count: hwc_and_sc_count(district_facilities),
+                       telemed_data: calculate_aggregates(district_facilities),
+                       users: sum_values(district_facilities, :users),
+                       facilities: district_facilities
+                         .select { |facility| %w[HWC SC].include? facility[:type] }
+                         .sort_by { |facility| facility[:name] }}
+                    }.sort_by { |district| district[:district] }}
+      }.sort_by { |state| state[:state] }
+    end
+
+    def high_bps(bps)
+      bps.hypertensive.uniq { |bp| bp[:patient_id] }
+    end
+
+    def high_sugars(blood_sugars)
+      blood_sugars.select(&:diabetic?).uniq { |bs| bs[:patient_id] }
+    end
+
+    def facility_measures(facility, p_start, p_end)
```

Comment:
Fetches per-facility information that we roll up for district and state level numbers